Automattic/wp-calypso
# Plans Subheader: fix subheader for tailored flows to respect the 'hideFreePlan' value

#### Proposed Changes

* For tailored flow the subheader text was not respecting the value of 'hideFreePlan' and this PR fixes that.

#### Testing Instructions
- Checkout this branch
- Run `yarn start`
- Test on the [newsletter flow](http://calypso.localhost:3000/setup?flow=newsletter)
- And also test on the [link in bio flow](http://calypso.localhost:3000/setup?flow=link-in-bio)
- When clicking on "Decide Later" or selecting the free wordpress domain in the domains page you should see subheader with a link <img width="803" alt="Screenshot 2022-09-02 at 18 08 17" src="https://user-images.githubusercontent.com/7000684/188194851-78a9de14-1234-4898-bbc0-071aacbb6a28.png">
- When selecting a paid domain it should NOT show a link to start with a free plan <img width="662" alt="Screenshot 2022-09-02 at 18 37 50" src="https://user-images.githubusercontent.com/7000684/188199440-f925810d-c325-4f5a-be6c-73952fd12a9b.png">


Related to #67330
